### ISSUE TITLE
Certain URLs not producing correct results on Lambda

### ISSUE THREAD

**Issue by kpeters-cbsi** [dougmoscrop/serverless-http#242]
First of all, I'm not sure where the problem ultimately lies, as this is being used to export Swagger docs via `swagger-ui-express`. Also, the problem seems limited to the context of AWS Lambda; I am unable to reproduce the problem locally. I have a question on StackOverflow, but I thought I'd post it here as well.

The Lambda function below will return the correct content with certain paths. With other paths it will return the standard Swagger UI HTML. As I said above, this only happens when actually running the function on Lambda. When I run it locally with the exact same request, it produces the correct output. 

I've attached a sample of a working and a non-working event.

Lambda code:
```
/** @format */
import 'source-map-support/register'
import express from 'express'
import serverless from 'serverless-http'
import swaggerUi from 'swagger-ui-express'
import { Callback, Context, Handler } from 'aws-lambda'
import { APIGatewayClient, GetExportCommand } from '@aws-sdk/client-api-gateway'

const app = express()
const apiGateway = new APIGatewayClient({})

export const handler: Handler = async (event, context, callback) => {
  const resource = event.resource
  const path = event.path
  await setup(event, context, callback)
  console.debug(`** REQUEST FOR ${path}`, { resource, event })
  console.debug('event', JSON.stringify(event, null, 2))
  console.debug('context', JSON.stringify(context, null, 2))
  const handlerFn = serverless(app, {
    request: (request: any, event: any, _context: any) => {
      console.debug(`Request URL as seen by Serverless: ${request.url}`)
      console.debug('Request headers', { headers: request.headers })
    },
  })
  console.debug('got handler', { handler: handlerFn })
  const ret = await handlerFn(event, context)
  console.debug('handler returned', { ret })
  return ret
}

const setup = async (event: any, context: Context, callback: Callback) => {
  const apiId = event.requestContext.apiId
  const stage = event.requestContext.stage

  let swaggerJson: swaggerUi.JsonObject
  try {
    if (stage === 'test-invoke-stage') {
      swaggerJson = await getSwaggerJson(apiId, 'prod')
    } else {
      swaggerJson = await getSwaggerJson(apiId, stage)
    }
  } catch (e) {
    console.error('Failed to retreive Swagger JSON', e)
    throw new Error('Failed to retreive Swagger JSON')
  }
  console.debug('Got Swagger doc object', { swaggerJson })
  app.use(
    '/swagger-simpler',
    swaggerUi.serve,
    swaggerUi.setup({ swaggerDoc: swaggerJson })
  )
  console.debug('app is set up')
}

const getSwaggerJson = async (
  restApiId: string,
  stageName: string
): Promise<swaggerUi.JsonObject> => {
  const params = {
    exportType: 'oas30',
    restApiId,
    stageName,
    accepts: 'application/json',
  }
  // console.debug('GetExportCommand', { params })
  const res = await apiGateway.send(new GetExportCommand(params))
  // console.debug('GetExportCommand successful', { res })
  let swaggerJson: string
  if (res.body) {
    swaggerJson = Buffer.from(res.body).toString()
  } else {
    throw new Error('Empty response body from GetExportCommand')
  }
  // console.debug('Got Swagger JSON', { swaggerJson })

  return JSON.parse(swaggerJson)
}
```

[sample-swagger-event-NOT-WORKING.json.txt](https://github.com/dougmoscrop/serverless-http/files/9611929/sample-swagger-event-NOT-WORKING.json.txt)
[sample-swagger-event-WORKING.json.txt](https://github.com/dougmoscrop/serverless-http/files/9611930/sample-swagger-event-WORKING.json.txt)

**Comment by dougmoscrop** [comment]
You shouldn't be initializing in the handler, you're actually registering middleware multiple times (if this isn't pseudocode).  It looks like you're trying to use a single function that can dynamically return different results depending on which APIg/stage you point at it, which isn't impossible but is certainly more complex.

If you can, I'd suggest deploying separate entire stages (are you using Serverless?) such that you pass in the stage name and the ApiId as environment variables, so you can do the setup initialization once.

If you can't do this, then you should still have setup() create an entirely new express() api for each configuration and return the handler, and then memoize it.

**Comment by kpeters-cbsi** [comment]
Thanks for the advice - I'll try that. The problem, however, turned out to be in the way `swagger-ui-express` was serving content and the way AWS CDK was deploying the Lambda. I wrote an [article](https://popefelix.medium.com/swagger-ui-via-aws-and-aws-cdk-a6b012c224e2) that gives the full explanation, but in a nutshell, CDK was bundling the Lambda and all of its dependencies into one big file, which messed up `swagger-ui-express` and I had to make it not do that. :)
